scipy/stats/_distribution_infrastructure.py

```diff
@@ -1999,7 +2073,10 @@ def _solve_bounded(self, f, p, *, bounds=None, params=None):
         p, xmin, xmax = np.broadcast_arrays(p, xmin, xmax)
         if not p.size:
             # might need to figure out result type based on p
-            return np.empty(p.shape, dtype=self._dtype)
+            res = _RichResult()
+            empty = np.empty(p.shape, dtype=self._dtype)
+            res.xl, res.x, res.xr = empty, empty, empty
+            res.fl, res.fr = empty, empty
```

Comment:
Kind of clumsy/annoying, but it allows the methods that call `_solve_bounded` to treat this as any other case. I figure it's better to special-case here rather than everwhere that calls this function.